main.py

```diff
@@ -58,52 +68,80 @@ def draw(state, a, b, colour):
         pygame.draw.rect(Screen, colour, (x, y, s, s))
 
 
-# noinspection PyShadowingNames
-def check_user_input(board, paused):
+def check_user_input(board, paused, gps, gps_limit):
     """Checks for user input and acts accordingly"""
     one_turn = False
     for event in pygame.event.get():
         x, y = pygame.mouse.get_pos()
-        a = maths.floor(x / config.Size) + config.Cushion
-        b = maths.floor(y / config.Size) + config.Cushion
+        a = x // config.Size + config.Cushion
+        b = y // config.Size + config.Cushion
         if pygame.key.get_pressed()[pygame.K_SPACE]:
-            if paused:
-                paused = False
-            else:
-                paused = True
+            paused = not paused
         if event.type == pygame.QUIT or pygame.key.get_pressed()[pygame.K_ESCAPE]:
             pygame.quit()
             import sys
             sys.exit(0)
         for key in range(pygame.K_1, pygame.K_8):
             if pygame.key.get_pressed()[key]:
                 board = preset.place(board, int(pygame.key.name(key)), a, b)
+                board = update_board(board)
+                draw_board(board)
+        if pygame.key.get_pressed()[pygame.K_f]:
+            gps_limit = not gps_limit
+            draw_gps_slider(((maths.log(gps, 10) + 1) / -3) * (config.EndOfSlider - config.StartOfSlider) +
+                            config.EndOfSlider, gps_limit)
         if pygame.key.get_pressed()[pygame.K_RIGHT]:
             one_turn = True
         if pygame.key.get_pressed()[pygame.K_RETURN]:
             board = clean_board()
+            board = update_board(board)
+            draw_board(board)
+            global Generations
```

Comment:
rename to `generations_since_start` and make it not global just as you did with other variables